### ISSUE TITLE
Adding a dapr integration for distributed deploy

### ISSUE THREAD

**Issue by sbiaudet** [OrchardCMS/OrchardCore#10016]
Hello,

I would like to be able to use orchardcore in a distributed environment with Dapr. Can we consider the creation of a Dapr module like the Redis module ?

I had imagined the following mapping:
   - IMessageBus based on dapr's PUB/SUB service 
   - IDistributedLock with the distributed lock algorithm (https://redis.io/topics/distlock) based on dapr's State storage service
   - ITagCache based on dapr's State storage service.

we can also consider an integration of dapr for a secure configuration with secret storage and this library https://www.nuget.org/packages/Dapr.Extensions.Configuration/

I would like to know if I can propose a pull request or a plug-in.

Sébastien

**Comment by Piedone** [comment]
Yes, if you'd benefit from a Dapr implementation of the various distributed services (since then we have more) then we'd be happy to take a look at a PR.

**Comment by hishamco** [comment]
Recently I created a module using Garnet in [Orchard Core Contrib](https://github.com/OrchardCoreContrib/OrchardCoreContrib.Modules/tree/main/src/OrchardCoreContrib.Garnet) but it's good to have Dapr too
